pennylane_lightning/lightning_qubit.py

```diff
@@ -531,9 +531,8 @@ def _process_jacobian_tape(self, tape, starting_state, use_device_state):
         all_params = 0
 
         for op_idx, tp in enumerate(trainable_params):
-            op, _ = tape.get_operation(
-                op_idx
-            )  # get op_idx-th operator among differentiable operators
+            # get op_idx-th operator among differentiable operators
+            op, _, _ = tape.get_operation(op_idx, return_op_index=True)
```

Comment:
If yes, I think setting `return_op_index` to `False` is more sensible.

Reply:
[See this entry on the deprecations page](https://github.com/PennyLaneAI/pennylane/blob/2efbb3b07d3665a2a208bed777c9120940a47bdd/doc/development/deprecations.rst?plain=1#L15)

This was a a breaking change that was raising a warning, so I just fixed it in this PR.

Reply:
Yes, I know there is a warning, but I believe that making `return_op_index=False` is sufficient to avoid breaking when the change comes, am I right? 
Keeping the warning, for now, is not a problem.

Reply:
Guess it should work.

Reply:
Specifying `return_op_index=False` will still give an error.

Reply:
I have no problem with keeping the warning for now.

Reply:
Is this going to impact the 0.30 release? If so, let's get rid of the warning.

Reply:
The old behaviour will be going away before the 0.30 release. So this change will need to be made before then anyway.